pkg/autoscaler/metrics/collector.go

```diff
@@ -385,6 +370,41 @@ func (c *collection) record(stat Stat) {
 	c.rpsPanicBuckets.Record(stat.Time, rps)
 }
 
+// startScraping starts scraping metrics.
+// This is only run from inside a lock, it can't race with a stopScraping call.
+func (c *collection) startScraping() {
+	c.stopCh = make(chan struct{})
```

Comment:
Right now this seems tight, but in general I'd like a panic/fatal in case of this being called twice on end (stop will already panic by trying to close the channel twice).